src/Service.GraphQLBuilder/GraphQLStoredProcedureBuilder.cs

```diff
@@ -122,5 +129,57 @@ public static FieldDefinitionNode GetDefaultResultFieldForStoredProcedure()
+        /// holds database schema metadata.
+        /// </summary>
+        /// <param name="defaultValueFromConfig">String representation of default value defined in runtime config.</param>
+        /// <param name="parameterDefinition">Database schema metadata for stored procedure parameter which include value and value type.</param>
+        /// <returns>Tuple where first item is the string representation of a GraphQLType (e.g. "Byte", "Int", "Decimal")
+        /// and the second item is the GraphQL {type}ValueNode </returns>
+        /// <exception cref="DataApiBuilderException">Raised when parameter casting fails due to unsupported type.</exception>
+        private static Tuple<string, IValueNode> ConvertValueToGraphQLType(string defaultValueFromConfig, ParameterDefinition parameterDefinition)
+        {
+            string paramValueType = SchemaConverter.GetGraphQLTypeFromSystemType(type: parameterDefinition.SystemType);
+
+            try
+            {
+                Tuple<string, IValueNode> valueNode = paramValueType switch
+                {
+                    UUID_TYPE => new(UUID_TYPE, new UuidType().ParseValue(Guid.Parse(defaultValueFromConfig))),
+                    BYTE_TYPE => new(BYTE_TYPE, new IntValueNode(byte.Parse(defaultValueFromConfig))),
+                    SHORT_TYPE => new(SHORT_TYPE, new IntValueNode(short.Parse(defaultValueFromConfig))),
+                    INT_TYPE => new(INT_TYPE, new IntValueNode(int.Parse(defaultValueFromConfig))),
+                    LONG_TYPE => new(LONG_TYPE, new IntValueNode(long.Parse(defaultValueFromConfig))),
+                    SINGLE_TYPE => new(SINGLE_TYPE, new SingleType().ParseValue(float.Parse(defaultValueFromConfig))),
+                    FLOAT_TYPE => new(FLOAT_TYPE, new FloatValueNode(double.Parse(defaultValueFromConfig))),
+                    DECIMAL_TYPE => new(DECIMAL_TYPE, new FloatValueNode(decimal.Parse(defaultValueFromConfig))),
+                    STRING_TYPE => new(STRING_TYPE, new StringValueNode(defaultValueFromConfig)),
+                    BOOLEAN_TYPE => new(BOOLEAN_TYPE, new BooleanValueNode(bool.Parse(defaultValueFromConfig))),
+                    DATETIME_TYPE => new(DATETIME_TYPE, new DateTimeType().ParseResult(
+                        DateTime.Parse(defaultValueFromConfig, DateTimeFormatInfo.InvariantInfo, DateTimeStyles.AssumeUniversal))),
+                    BYTEARRAY_TYPE => new(BYTEARRAY_TYPE, new ByteArrayType().ParseValue(Convert.FromBase64String(defaultValueFromConfig))),
+                    LOCALTIME_TYPE => new(LOCALTIME_TYPE, new LocalTimeType().ParseResult(LocalTimePattern.ExtendedIso.Parse(defaultValueFromConfig).Value)),
+                    _ => throw new NotSupportedException(message: $"The {defaultValueFromConfig} parameter's value type [{paramValueType}] is not supported.")
+                };
+
+                return valueNode;
+            }
+            catch (Exception error) when (
```

Comment:
What happens if we get an exception that is not of one of these error types?

Reply:
I tried it out by stepping through debugger and throwing a different type of exception. This resulted in hard to decipher stack trace and uncaught error during dab startup. 
![image](https://github.com/Azure/data-api-builder/assets/6414189/3d4cf7e4-552e-4702-884f-eae47586183b)

Good question and I modified this to no longer filter on certain exception types, because any issue with processing the value here should result in an easier to read exception.

When this catches all exception types, the message is easier to read: 
![image](https://github.com/Azure/data-api-builder/assets/6414189/a0939699-98c6-45e0-bd2e-a22fb704c418)

though i do admit we should probably be catching this exception elsewhere too so that we can eliminate long unnecessary stack traces printed to the console.

Reply:
my intent with filtering certain exceptions is that I'd only want to emit the "the <param_name> parameter's value type [Type] is not supported." message when an actual parse error occurred. Other exceptions wouldn't necessarily fit to this message.

Reply:
updated to no longer filter on specific exception type.

Reply:
thanks, good info here!